modules/local_storage/src/LocalStockServiceConfig.php

```diff
@@ -34,17 +35,17 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager) {
   /**
    * {@inheritdoc}
    *
-   * @ToDo Do we really need a primary transaction location. If so users need a way to set this.
    */
-  public function getPrimaryTransactionLocation(PurchasableEntityInterface $entity, $quantity) {
-    $locations = $this->getLocationList($entity);
+  public function getTransactionLocation(Context $context, PurchasableEntityInterface $entity, $quantity) {
+    // location.
+    $locations = $this->getAvailabilityLocations($context, $entity);
     return empty($locations) ? NULL : array_shift($locations);
```

Comment:
I haven't looked at the resolver. Open to using it. What would be the advantage of using it over inheriting from the existing one. A new implementation will probably need a configuration screen to go with any changes. I think what the best investment would be to make current configuration more flexible so adding a checkbox for a new option "Allocate stock for first location with availability" would be great.